src/software/new_geom/ray.h

```diff
@@ -0,0 +1,85 @@
+#pragma once
+
+#include "software/geom/line.h"
+#include "software/geom/point.h"
+
+class Ray final
+{
+   public:
+    /**
+     * Creates a degenerate Ray at (0, 0)
+     */
+    inline explicit constexpr Ray() {}
+
+    /**
+     * Creates a Ray with a start point and a direction
+     *
+     * @param start the starting point of Ray
+     * @param direction the direction of Ray
+     */
+    inline explicit Ray(const Point& start, const Vector& direction)
+        : start_(start), direction_(direction)
+    {
+    }
+
+    /**
+     * Returns the start point of this Ray
+     *
+     * @return the start point of this Ray
+     */
+    inline Point getStart() const
+    {
+        return start_;
+    }
+
+    /**
+     * Returns the direction of this Ray
+     *
+     * @return the direction of this Ray
+     */
+    inline Vector getDirection() const
+    {
+        return direction_;
+    }
+
+    /**
+     * Sets the start point of this Ray
+     *
+     * @param point the new start point
+     */
+    inline void setStart(const Point& start)
+    {
+        start_ = start;
+    }
+
+    /**
+     * Sets the direction of this Ray
+     *
+     * @param direction the new direction
+     */
+    inline void setDirection(const Vector& direction)
```

Comment:
We should probably have another version of this function that takes an `Angle` as the direction.

Reply:
Addressed in fbdf4f3